components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java

```diff
@@ -84,7 +93,14 @@ public void completedServerStartup() {
                 new APIMgtGatewayCacheMessageListener());
         jmsTransportHandlerForEventHub
                 .subscribeForJmsEvents(APIConstants.TopicNames.TOPIC_NOTIFICATION, new GatewayJMSMessageListener());
+    }
 
+    private void deployAPIsInSyncMode() {
+        isAPIsDeployedInSyncMode = deployArtifactsAtStartup();
+        if (!isAPIsDeployedInSyncMode) {
+            log.error("Unable to deploy synapse artifacts at gateway");
+            deployAPIsInSyncMode();
```

Comment:
I feel we need to configure a retry limit rather than retrying indefinitely. With infinite retrying there'd be a constant load on the underlying storage and a few failed APIs can exhaust the storage fairly quickly.

Reply:
https://github.com/wso2/carbon-apimgt/pull/8945/commits/ae40fd8c53d3812a64ea9c97d69bb8600ff81a52